dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/VelocityRequestWrapper.java

```diff
@@ -1,347 +1,149 @@
 package com.dotcms.rendering.velocity.viewtools;
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.UnsupportedEncodingException;
-import java.security.Principal;
-import java.util.Collection;
-import java.util.Enumeration;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Set;
 
-import javax.servlet.AsyncContext;
-import javax.servlet.DispatcherType;
-import javax.servlet.RequestDispatcher;
+import java.util.Set;
 import javax.servlet.ServletContext;
 import javax.servlet.ServletException;
-import javax.servlet.ServletInputStream;
-import javax.servlet.ServletRequest;
-import javax.servlet.ServletResponse;
-import javax.servlet.http.*;
-
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
 import com.dotmarketing.util.Config;
 import com.dotmarketing.util.SecurityLogger;
 import com.dotmarketing.util.UtilMethods;
 import com.google.common.collect.ImmutableSet;
 import com.liferay.portal.util.WebKeys;
 import com.liferay.util.Xss;
 
-public class VelocityRequestWrapper extends javax.servlet.http.HttpServletRequestWrapper  {
+/**
+ * This class wraps the incoming HttpServletRequest to provide security and to allow uris to be
+ * overridden
+ * 
+ * @author will
+ *
+ */
+public class VelocityRequestWrapper extends javax.servlet.http.HttpServletRequestWrapper {
```

Comment:
Codacy found an issue: [A class which only has private constructors should be final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5831320)